cg/apps/orderform/excel_orderform_parser.py

```diff
@@ -171,50 +171,26 @@ def parse_data_analysis(self) -> str:
 
         return data_analyses.pop().lower().replace(" ", "-")
 
-    def is_from_orderform_without_data_delivery(self, data_delivery: str) -> bool:
-        return data_delivery == self.NO_VALUE
-
-    def default_delivery_type(self, project_type: OrderType) -> str:
-        """Returns the default delivery type for a project type"""
-
-        if project_type == OrderType.MICROSALT and self.parse_data_analysis() in [
-            "custom",
-            "fastq",
-        ]:
-            return DataDelivery.FASTQ_QC
-
-        raise OrderFormError("Could not determine value for Data Delivery")
-
-    def get_data_delivery(self, project_type: OrderType) -> str:
+    def get_data_delivery(self) -> str:
         """Determine the order_data delivery type."""
 
         data_delivery: str = self.parse_data_delivery()
 
-        if self.is_from_orderform_without_data_delivery(data_delivery):
-            return self.default_delivery_type(project_type)
-
-        if data_delivery == "fastq-qc":
-            return DataDelivery.FASTQ_QC
-        if data_delivery == "fastq-qc-+-analysis":
-            return DataDelivery.FASTQ_QC_ANALYSIS
-        if data_delivery == "fastq-qc-+-analysis-+-cram":
-            return DataDelivery.FASTQ_QC_ANALYSIS_CRAM
-        if data_delivery == "fastq-qc-+-analysis-+-cram-+-scout":
-            return DataDelivery.FASTQ_QC_ANALYSIS_CRAM_SCOUT
-
         try:
             return DataDelivery(data_delivery)
-        except ValueError:
-            raise OrderFormError(f"Unsupported Data Delivery: {data_delivery}")
+        except ValueError as e:
+            raise OrderFormError(f"Unsupported Data Delivery: {data_delivery}") from e
 
     def parse_data_delivery(self) -> str:
 
-        data_deliveries = {sample.data_delivery or self.NO_VALUE for sample in self.samples}
+        data_deliveries: Set[str] = {
+            sample.data_delivery or self.NO_VALUE for sample in self.samples
+        }
 
         if len(data_deliveries) > 1:
             raise OrderFormError(f"mixed 'Data Delivery' types: {', '.join(data_deliveries)}")
 
-        return data_deliveries.pop().lower().replace(" ", "-")
+        return data_deliveries.pop().lower().replace(" + ", "-").replace(" ", "_")
```

Comment:
@henrikstranneheim Maybe not the nicest line, but it simplifies a lot. Do you think I should split this one up?

Reply:
Created a new function taking care of the string manipulation and created a simple test for the `get_data_delivery` method.